File: backend/src/main/java/com/vn/tripfinity/backend/dto/auth/LoginResponse.java
package com.vn.tripfinity.backend.dto.auth;

import lombok.Data;

@Data
public class LoginResponse {
    private String token;
    private String type = "Bearer";
    private Integer userId;
    private String name;
    private String email;

    public LoginResponse(String token, Integer userId, String name, String email) {
        this.token = token;
        this.userId = userId;
        this.name = name;
        this.email = email;
    }
}

File: app/lib/dto/auth/login_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'login_response.g.dart';

@JsonSerializable()
class LoginResponse {
  final String token;
  final String type;
  final int userId;
  final String name;
  final String email;

  LoginResponse({
    required this.token,
    this.type = 'Bearer',
    required this.userId,
    required this.name,
    required this.email,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) =>
      _$LoginResponseFromJson(json);

  Map<String, dynamic> toJson() => _$LoginResponseToJson(this);
}
